scipy/special/tests/test_basic.py

```diff
@@ -79,8 +79,8 @@ def test_binom(self):
         assert_func_equal(cephes.binom, rknown.ravel(), nk, rtol=1e-13)
 
         # Test branches in implementation
-        rng = np.random.RandomState(1234)
-        n = np.r_[np.arange(-7, 30), 1000*rng.rand(30) - 500]
+        np.random.seed(1234)
+        n = np.r_[np.arange(-7, 30), 1000*np.random.rand(30) - 500]
```

Comment:
I think you picked up a bunch of changes here that revert the random number handling back to using global state - looks like a mistake?